providers/keycloak_oidc_test.go

```diff
@@ -0,0 +1,176 @@
+package providers
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/coreos/go-oidc"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	accessTokenHeader    = "ewogICJhbGciOiAiUlMyNTYiLAogICJ0eXAiOiAiSldUIgp9"
+	accessTokenPayload   = "eyJyZXNvdXJjZV9hY2Nlc3MiOiB7ImRlZmF1bHQiOiB7InJvbGVzIjogWyJyZWFkIl19fX0"
+	accessTokenSignature = "dyt0CoTl4WoVjAHI9Q_CwSKhl6d_9rhM3NrXuJttkao"
+)
+
+type DummyKeySet struct{}
+
+func getAccessToken() string {
+	return fmt.Sprintf("%s.%s.%s", accessTokenHeader, accessTokenPayload, accessTokenSignature)
+}
+
+func (DummyKeySet) VerifySignature(_ context.Context, _ string) (payload []byte, err error) {
+	p, _ := base64.RawURLEncoding.DecodeString(accessTokenPayload)
+	return p, nil
+}
+
+func newTestKeycloakOIDCSetup(body []byte) (*httptest.Server, *KeycloakOIDCProvider) {
+	redeemUrl, server := newOIDCServer(body)
+	provider := newKeycloakOIDCProvider(redeemUrl)
+	return server, provider
+}
+
+func newKeycloakOIDCProvider(serverURL *url.URL) *KeycloakOIDCProvider {
```

Comment:
For any new test files, we are looking to migrate everything to using Ginkgo BDD test suites.

As a reference, you can keep most of these table driven tests, here's a sample from @JoelSpeed using the `DescribeTable` functionality in Ginkgo: https://github.com/oauth2-proxy/oauth2-proxy/pull/1142/files

If you need any references, just look at any of @JoelSpeed 's PRs, he's a Ginkgo testing wizard 😄

Reply:
Thanks, yes I should rewrite them using the newer style. Looks like some examples are already in the repo.